root/components/common-macros.tt

```diff
@@ -895,18 +895,13 @@ END ~%]
 [%- END -%]
 
 [%~ MACRO show_image BLOCK -%]
-    [%- IF image -%]
-        [%- INCLUDE 'components/commons-image.tt' -%]
-    [%- ELSIF image_url -%]
-        <span style="display:none" id="image-insertion-point"></span>
-        <script type="text/javascript">
-            $.get('[% image_url %]',
-                  function (data) {
-                      $('#image-insertion-point').replaceWith(data);
-                  },
-                  'html');
-        </script>
-    [%- END -%]
+    <div id="commons-image">
+        [%~ React.embed(c, 'root/components/CommonsImage', { image => image }) ~%]
```

Comment:
If `image` was loaded from the cache and is defined here, this will render completely on the server. We don't embed the `commons-image.js` bundle in that case either, due to the `UNLESS image` below.